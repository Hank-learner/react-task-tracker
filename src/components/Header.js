import PropTypes from 'prop-types';
import {useLocation} from 'react-router-dom'
import Button from './Button'

const Header = ({ title, onToggleShowAddTask, showAddTask }) => {
    const location = useLocation()

    return (
        <header className='header' id='header'>
            <h1>{title}</h1>
            {location.pathname ==='/' &&
                <Button
                    enabled = {true}
                    color={showAddTask ? 'red' : 'green'}
                    text={showAddTask ? 'Close' : 'Add'}
                    onClick={onToggleShowAddTask}
                    dataTestID='add-task'
                />
            }
        </header>
    )
}

Header.defaultProps = {
    title:"Task Tracker",
}

Header.propTypes = {
    title: PropTypes.string.isRequired,
}

// const headingStyle = {
//     color: "red",
//     backgroundColor: "black"
// }

//class
// class Header extends React.Component{
//     constructor(props) {
//         super(props)
//         this.state = {
//             count: 0
//         };
//     }
//     render() {
//         return (
//             <p>{this.state.count}</p>
//         );
//     }
// }

//useRef hook - doesnt re render like setstate
// function App(){
// const myBtn = useRef(null)
// const clickIt = () => myBtn.current.click()
// return (<button ref={myBtn}></button>)
//}


export default Header;
