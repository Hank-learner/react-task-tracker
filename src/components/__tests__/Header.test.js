import renderer from 'react-test-renderer';
import Header from '../Header';

// const validProps = {

// }

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useLocation: () => ({
        pathname: "localhost:3000/"
    })
}));

describe('Header snapshot test', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<Header />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});