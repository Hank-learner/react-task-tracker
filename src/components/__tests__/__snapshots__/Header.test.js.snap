// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot test renders correctly 1`] = `
<header
  className="header"
  id="header"
>
  <h1>
    Task Tracker
  </h1>
</header>
`;
